3rdparty/obiwarp/mat.h

```diff
@@ -22,458 +22,396 @@ class MatD;
 class MatF {
 
     public:
-        // length
-        int _m;
-        int _n;
-        VecF _dat;
-        // Constructors:
-        MatF();
-        MatF(int m, int n);
-        MatF(int m, int n, const float &val);
-       
-
-        // (copied from vec.h)
-        // if (shallow == 1 (true)) then no memory is deleted upon destruction
-        // if (shallow == 0 (false)) then delete[] is called
-        // FOR THIS CONSTRUCTOR ONLY, there is no DEEP copying, EVER!
-        MatF(int m, int n, float *arr, bool shallow=0);
-
-        // (copied from vec.h)
-        // if (shallow == 0 (false)) a DEEP copy is made of the data
-        // if (shallow == 1 (true)) a copy of the pointer is made
-        // if (shallow) then no memory is released upon destruction
-        // shallow is used for a quick copy with which to work 
-        MatF(const MatF &A, bool shallow=0);
-
-        operator float*() { return (float*)_dat; }
-        operator const float*() { return (float*)_dat; }
-        float* pointer() { return (float*)_dat; }
-        float* pointer(int m) { return &_dat[m*_n]; }
-        // creates vec objects 
+    // length
+    int _m;
+    int _n;
+    VecF _dat;
+    // Constructors:
+    MatF();
+    MatF(int m, int n);
+    MatF(int m, int n, const float &val);
+
+
+    // (copied from vec.h)
+    // if (shallow == 1 (true)) then no memory is deleted upon destruction
```

Comment:
Would be great if the method docs could be cleaned up. `shallow` is not an arg anymore

Reply:
Consider it a suggestion not a requirement

Reply:
Oh I had it as a task in my head. Maybe it got lost in all the excitement. Thanks for reminding. Will make sure the docs do not have any incorrect information.